cmd/server/cadence/server.go

```diff
@@ -203,10 +205,13 @@ func (s *server) startService() common.Daemon {
 	)
 
 	params.ArchiverProvider = provider.NewArchiverProvider(s.cfg.Archival.History.Provider, s.cfg.Archival.Visibility.Provider)
-
 	params.PersistenceConfig.TransactionSizeLimit = dc.GetIntProperty(dynamicconfig.TransactionSizeLimit, common.DefaultTransactionSizeLimit)
-
 	params.Authorizer = authorization.NewNopAuthorizer()
+	params.BlobstoreClient, err = filestore.NewFilestoreClient(s.cfg.Blobstore.Filestore)
```

Comment:
could we move this to worker/server.go? The other three servers don't need this blob store client.

Reply:
They might. I think this is a common utility just like archiver.

Reply:
I agree that blobstore can be a common utility but I think we need to think more about how to config/initialize it. Right now it seems like there's only one global config for it. I mean the entire system can use only one blobstore implementation with a certain config right now. But maybe in the future there will be use cases where different components need different blobstore implementations. Take a step back, even within an implementation, I think different configs might be needed. For example, archiver want to upload history to a path(bucket) with very long retention period, while the bucket for scanner only need a very short retention time.

Reply:
Followed up offline on this - will create ticket around this.